homeassistant/components/gaposa/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""The Gaposa integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from pygaposa import FirebaseAuthException, Gaposa, GaposaAuthException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_PASSWORD, CONF_USERNAME, DOMAIN, UPDATE_INTERVAL
+from .coordinator import DataUpdateCoordinatorGaposa
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.COVER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Gaposa from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN].setdefault(entry.entry_id, {})
```

Comment:
please extend the typing of `ConfigEntry` and store the data you would store in `hass.data` in `entry.runtime_data`